helix-term/src/ui/mod.rs

```diff
@@ -94,10 +108,13 @@ pub fn regex_prompt(
                         false
                     };
 
-                    match RegexBuilder::new(input)
-                        .case_insensitive(case_insensitive)
-                        .multi_line(true)
-                        .build()
+                    match rope::RegexBuilder::new()
+                        .syntax(
+                            rope::Config::new()
+                                .case_insensitive(case_insensitive)
+                                .multi_line(true),
```

Comment:
With this branch it seems like the `^` anchor doesn't have an effect. Is that intentional or a regression?

Reply:
Hmm it should work it worked before when I was trying stuff but I will look at testing that again later.  Do you have somekind of more detailed reproduction case? Was it global search or local search or `s`. With `s` I remember it working at least.

Reply:
Oh whoops, it still works at the beginnings of lines but not within selections. For example you can select the `p` in `picture` in https://github.com/helix-editor/helix/blob/cf4492174d0ee27bd3c73a5fa57fe3a26aa064be/README.md?plain=1#L4 with `^.` or `^p` on master but not this branch. Same with `$`: it works at the ends of lines but not to anchor to the end of a selection

Reply:
so this one is kind of on purpose.

The reason is that in the past we created a string from the selection and then ran the regex of that so the start of the selection was the start of the file from the perspective of regex.

The new API allows us to instead correctly run this as a search starting at <selection start> within the full rope. In that case `^` does not count as file start anymore. I could change this back in theory but I am not sure I should.

The effect of the way it worked in the past is that the start/end of selection did not just match anchor but also word boundaries. For example `x|foo|x` `s\bfoo\b` matches on master but not on this branch.

I think the concept of anchoring the search at the start/end of the selection is useful. I think that we should use some other way to achieve that tough that doens't break every other lookaround assertion and also doens't match line start/end. It makes sense that the start/end of a file match as line starts/ends since they really are the start/end of the first/last line but the same doens't make sense for selection start/end IMO

Reply:
Ah ok that makes sense. I think it would be useful to have anchors for the selection start and end but I'm not attached to re-using `^`/`$`. I might be misremembering but I believe these are `\A` and `\Z` in Kakoune

Reply:
I think the issue will ultimately be that we need to parse these ourselves and then get support for something like this (anchor at search start/end instead of haystack start/end) upstream. I could just add that to regex-cursor but I want to upstream that eventually so I would rather not add not features without buyin from upstream. 

I will try to see if I can either find something avout this upstream or create an issue

Reply:
ah nevermind this is only supported upstream (anchored searches)i I just need to figure out the syntax

Reply:
I am looking at implementing this upstream but it won't be easy and take a while: https://github.com/rust-lang/regex/discussions/1161#discussioncomment-8328965.

I think in the meantime while loosing this accidental feature is unfortunate its better than the matching being buggy. Would you be ok with keeping the behaviour from this branch until this is properly supported in the regex engine?

Reply:
Yep I think it's ok to lose it for now. I haven't used it in recent memory and only noticed it when looking into #9459